### ISSUE TITLE
Exported GPX file contains invalid attribute "id" for "finder" tag

### ISSUE THREAD

**Issue by Dunkelmann** [cgeo/cgeo#3724]
Exported gpx files have a tag with the following format

> <groundspeak:finder id=""> FinderNameHere</groundspeak:finder>

On importing this file, it is possible for import programs to complain at the null attribute value for id.

Suggestion: to remove this possibility set a 0 value for the "id" attribute for this tag or insert a valid finder id value.

E.g.

> <groundspeak:finder id="0"> FinderNameHere</groundspeak:finder>

**Comment by Lineflyer** [comment]
My conclusion of this would be that we don't have other options than to hand over a empty value here.
So closing this, as we can not fix it and handing over "0" would be incorrect as explained above.
Please feel free to reopen if you disagree.

**Comment by ecornely** [comment]
Reading the XSD I would have imagined that it is impossible to have founder without having its ID.
You can have no founder minOccurs=0 but if you have one it has to have an ID in form of a string.

Reading the informations provided as JSON by geocaching.com (http://pastebin.com/KeiHUY2V), I think we should take the JSON path "AccountID" as founder ID.

I never understood exactly why in GPX format we need internal IDs coming from groundspeak but at least that's how I would fill the attribute. (Someone with a premium account could check wich field is taken)

Here would be a partial patch http://pastebin.com/5SrWjJFA adding a new field to LogEntry, using it in the GpxSerializer and filling it in the GCParser.
I already see that it also requires the field to be saved and read correctly in cgeo.geocaching.DataStore#loadLogs, to be done in cgeo.geocaching.connector.oc.OkapiClient#parseLogs and so on ... but i just started Android development a few days ago so I don't feel comfortable to include this in the code...

What do you think about that ?

**Comment by Lineflyer** [comment]
Reopening for further clarification

**Comment by SammysHP** [comment]
Do we need that ID somewhere else? If not, why clutter the DB? Apropos DB: I cannot see where you write the value to the database.

**Comment by ecornely** [comment]
In my patch I don't. I stopped my investigation when I realized the large impact it would have to add that information to the LogEntry class.

Personnaly I can live with an ID always set to 0 so that we comply with the XSD of groundspeak namespace (though an id should probably be unique). On the version 1.0.1 (http://www.groundspeak.com/cache/1/0/1/cache.xsd), it's clearly stated that this owner id *Owner ID corresponds to an account on Geocaching.com*.

For me cgeo and others probably will never need that geocaching.com internal ID. The question is do we wnat to completely respect the xsd and provide the right ID (which is easy to get : see my patch) or do we just want our gpx to be xsd compliant ?
